### PR TITLE
TalentSearch - New Added Skills component

### LINKED ISSUE #1477: My Skills - Added Skills component
# Description
- This is the section of the Experience form which shows "Skills attached to this experience". This shows the skills pill list format. 
- Skill pills should be clickable, which calls a removeSkill handler function passed into the component
- It should show some null state text when the list is empty
- If the list has more than 6 skills, show warning text in a yellow box

Question: what's the null state?

**[Experience Form](https://www.figma.com/proto/PrvXEDac0HVp7dF7SLWVFm/Candidate-profile-interfaces---V5.0.0?node-id=2340%3A4352&scaling=min-zoom&page-id=0%3A1&starting-point-node-id=1%3A2)**

Acceptance Criteria:
- [x] Create AddedSkills component with correct format and styling
- [x] A storybook story to view the component
- [x] Shows null value when there are no skills
- [x] Shows "That's a lot of skills" text when 6 or more in list
- [x] Pills are clickable
- [ ] Simple Jest tests

### PR DESCRIPTION
**Pull request by petertgiles**
This branch adds a new component for the Added Skills section.
- Add new AddedSkills component
- Restore old Alert component for use in new component
- Add storybook stories
- Add jest tests
- Add jest configuration option to allow use of common components in talentsearch
- Add storybook configuration to allow story file to be near component file
- Move Yoni's SearchBar component under the Skills section in storybook
- Tweak padding on Chip and add roles for easier testing
- Cherrypick Yoni's fakeSkills generator

Closes #1477

### CODE REVIEW

File: frontend/common/src/components/Alert/Alert.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import React, { DetailedHTMLProps } from "react";
+
+const DismissBtn: React.FC<
```
Comment (tristan-orourke) [inline frontend/common/src/components/Alert/Alert.tsx:3]: Should the default DismissBtn have some additional styling? Or use our Button component?

Comment (petertgiles) [reply frontend/common/src/components/Alert/Alert.tsx]: Ah well, I tried to reuse an existing component from the commit history but I'm not sure I want to spend time making it work for this situation.  This message doesn't even have a dismiss button in the design.  I've removed the Alert component and just used a regular div instead.

File: frontend/common/src/fakeData/fakeSkills.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      fr: `FR ${faker.lorem.paragraph()}`,
+    },
+    families: [],
+    keywords: [],
+  };
+};
+
+export default (): Skill[] => {
+  faker.seed(0); // repeatable results
+  return [...Array(10)].map(() => generateSkill());
```
Comment (tristan-orourke) [inline frontend/common/src/fakeData/fakeSkills.ts:25]: `list.map(() => generateSkill())` is equivalent to `list.map(generateSkill)`, with one less function definition.

File: frontend/talentsearch/resources/js/components/skills/AddedSkills/AddedSkills.test.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+describe("AddedSkills tests", () => {
+  test("If the skills collection is empty then the null message is displayed", () => {
+    renderContainer([], jest.fn());
+    const message = screen.getByText(
+      "There are no skills attached to this experience yet.",
+      { exact: false },
+    );
+    expect(message).toBeTruthy();
+  });
+  test("If the skills collection is not empty then the matching number or chips should be rendered", async () => {
```
Comment (tristan-orourke) [inline frontend/talentsearch/resources/js/components/skills/AddedSkills/AddedSkills.test.tsx:32]:
```suggestion
  test("If the skills collection is not empty then the matching number of chips should be rendered", async () => {
```

**Review by tristan-orourke — APPROVED** [review]
Looks great! I left a typo suggestion, and there's a couple merge conflicts, but I'm happy if you merge it in as soon as you deal with the conflicts.

File: frontend/common/src/fakeData/fakeSkills.ts (hunk 2 above)
Comment (petertgiles) [reply frontend/common/src/fakeData/fakeSkills.ts]: Looks like this was changed in James's PR and was part of the merge conflict

File: frontend/talentsearch/resources/js/components/skills/AddedSkills/AddedSkills.test.tsx (hunk 3 above)
Comment (petertgiles) [reply frontend/talentsearch/resources/js/components/skills/AddedSkills/AddedSkills.test.tsx:32]: Good catch!

**Pull request merged**